Common/src/geometry/CGeometry.cpp

```diff
@@ -3942,5 +3942,36 @@ void CGeometry::ComputeWallDistance(const CConfig* const* config_container, CGeo
         geometry->SetWallDistance(0.0);
       }
     }
+    /*--- Otherwise, set wall roughnesses. ---*/
+    if(!allEmpty){
+      /*--- Store all wall roughnesses in a common data structure. ---*/
+      // [iZone][iMarker] -> roughness, for this rank
+      auto roughness_f =
+        make_pair( nZone, [config_container,geometry_container,iInst](unsigned long iZone){
+          const CConfig* config = config_container[iZone];
+          const auto nMarker = geometry_container[iZone][iInst][MESH_0]->GetnMarker();
+
+          return make_pair( nMarker, [config](unsigned long iMarker)->su2double{
+            if(config->GetWallRoughnessProperties(config->GetMarker_All_TagBound(iMarker)).second!=0.0)
+            std::cout << "roughness!=0 in roughness_local construction: " << config->GetWallRoughnessProperties(config->GetMarker_All_TagBound(iMarker)).second << "\n";
+            return config->GetWallRoughnessProperties(config->GetMarker_All_TagBound(iMarker)).second;
+          });
+        });
+      NdFlattener<2> roughness_local(roughness_f);
+      NdFlattener<2> const& roughness_local_c = roughness_local;
+      std::cout << "roughness_local[0][0]=" << roughness_local[0][0] << "\n";
+      std::cout << "const roughness_local[0][0]=" << roughness_local_c[0][0] << "\n";
+      // [rank][iZone][iMarker] -> roughness
+      NdFlattener<3> roughness_global(Nd_MPI_Environment(), roughness_local);
+      std::cout << "roughness_global[0][0][0]=" << roughness_global[0][0][0] << "\n";
+      NdFlattener<3> const& roughness_global_c = roughness_global;
+      std::cout << "const roughness_global[0][0][0]=" << roughness_global_c[0][0][0] << "\n";
```

Comment:
Output in the serial AD testcase discadj_incomp_cylinder:

    ...
    Computing wall distances.
    roughness_local[0][0]=0
    const roughness_local[0][0]=0
    roughness_global[0][0][0]=-nan
    const roughness_global[0][0][0]=-nan
    GlobalIndex.size()=14576
    SetWallRoughness 1: set iPoint=0 roughness=-nan
    ...